WordPress/openverse-frontend
# Prevent clearing of the search bar value when homepage is hydrated

## Fixes


Fixes #1831 by @sarayourfriend

## Description


To prevent clearing out of the search bar on hydration, this PR makes the search bar an `uncontrolled` input. Instead of controlling the value of the search bar in the Vue code, we get the HTML input `value` on the form submit.
This component is used on the home page and the 404 page.

Other changes:
- `Oops` svg converted to an inline `svg`.
- `standalone` styles removed from the normal search bar component (`VSearchBarOld`).
- `ClientOnly` component wrapper removed from the search type popup inside the homepage search bar. Instead, it is shown conditionally using `v-show`. This prevents the client-server mismatch and makes the component loading faster.

## Testing Instructions

The search bar on both the home page and the 404 pages should work correctly.
Try throttling the network using dev tools, reloading the homepage, and entering something. The search term should not be deleted when the homepage is hydrated.

## Checklist


[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin


<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>
